### PR TITLE
Simplify serial dropdown list and include CRSF

### PR DESCRIPTION
**Pull request by ctzsnooze**
For discussion:

Update to the Serial Rx Dropdown menu.  Current messages overflow the available space and seem unnecessarily verbose.  

CRSF is a commonly used protocol used today, whereas Spectrum satellites are very uncommon, yet `SPEKSAT` has been the first in the list for a long time.  

Changes:

|    Old   |   New   |
|----|----|
| PPM RX input | PPM/CPPM (single wire) |
| Serial-based receiver (SPEKSAT, SBUS, SUMD | Serial (CRSF, SBUS, Ghost... |
| PWM RX input (one wire per channel) | PWM (one wire per channel) |
| MSP RX input (control via MSP port) | MSP (control via MSP port) |
| SPI RX support | SPI (e.g. built-in Rx) |

New appearance
Screen Shot 2021-11-27 at 12 21 04

Old appearance...
Screen Shot 2021-11-27 at 11 45 37
Old list:
Screen Shot 2021-11-27 at 12 07 45

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by limonspb — DISMISSED** [review]

File: locales/en/messages.json
Comment (Asizon) [inline locales/en/messages.json:1123]: The correct name here is "SPI Rx", this is how named in most of tutorials, and users knows with this name.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

File: locales/en/messages.json
Comment (ctzsnooze) [reply locales/en/messages.json]: Indeed, however the "Rx" part seems unnecessary, given that this is the connection mode list for the Rx.  
SPI is indeed used for other things like gyro, but here we know we are referring to SPI Rx only, it can be no other thing.
None the less I will add Rx back, it just seems unnecessary waste of screen.
It would be like adding Rx after CRSF, or after PWM.
They are all Rx modes.

File: locales/en/messages.json
Comment (mikeller) [inline locales/en/messages.json:1048]: This is incorrect. This protocol uses PPM, not PWM.

File: locales/en/messages.json
Comment (mikeller) [inline locales/en/messages.json:1057]: No issue with adding more proprietary protocols, but I think we should leave the open source alternatives like  SUMD in the list.

**Review by mikeller — CHANGES_REQUESTED** [review]

File: locales/en/messages.json
Comment (klutvott123) [reply locales/en/messages.json:1057]: Do we even need to list any protocols here? All of them will appear in the provider dropdown when serial is selected.

File: locales/en/messages.json
Comment (ctzsnooze) [reply locales/en/messages.json]: Thanks, yes.  PPM is a series of pulses and their offset from the clock provides a PWM equivalent, so many people think of it as 'serial PWM', though it isn't, strictly.  I was trying to differentiate it from PWM.
I wonder if the list order should be changed to put serial at the top and PPM above CPPM?
Those old forms are archaic these days.

File: locales/en/messages.json
Comment (ctzsnooze) [reply locales/en/messages.json:1057]: The only reason they are there is as an example, to help the noob decide which to choose.  
We don't really **need** anything other than the protocol name imho.  
However perhaps for beginners it is easier to give a kind of pseudo-explanation.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

File: locales/en/messages.json
Comment (limonspb) [reply locales/en/messages.json:1057]: I would agree here with @ctzsnooze - it's for most people to make the dropdown choice easier when they set up drones.

**Review by limonspb — APPROVED** [review]

**Review by limonspb — CHANGES_REQUESTED** [review]
Please squash :)

**Comment by ctzsnooze** [comment]
Closing - replaced by #2671

**Pull request closed without merging**
